codice/ddf
# DDF-4227 Upgrade to Karaf 4.2.1

#### What does this PR do?
Note: This is needed for upgrading to Java 11.
- Upgrades to Karaf 4.2.1
- Removes our custom "fork" of ConfigAdmin's ConfigurationManager, since the workaround is no longer needed in the current version of Felix ConfigAdmin

Known issue:
Client scripts not working properly on Windows (https://issues.apache.org/jira/browse/KARAF-5996)
#### Who is reviewing it? 
@blen-desta @bakejeyner @emmberk @Lambeaux 

#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@coyotesqrl 
#### How should this be tested?
Note: These should be tested across OS's
- Manually test basic functionality of DDF.
- Manually modify some properties in custom.system.properties, restart, and verify the properties are updated.
- Drop .cfg and .config files in the /etc directory, and confirm that the configurations are installed successfully.
- Confirm that the ./bin scripts work as expected.
- **For this Karaf upgrade only**: Verify passwords in config files in the /etc directory are being encrypted. Also verify all config files in the bundle cache have encrypted values. (Found at `<root>/data/cache/org.eclipse.osgi/<id>` where `<id>` is the bundle ID of configadmin. It should be 14 as of this change.)
- Walkthrough the Hardening Guide.
- Confirm the command line is still functional:
  - waitForReady
  - profile:* commands
  - migration:export and migration:import
  - lna
  - bundle:list, bundle:watch, bundle:headers
  - feature:list, feature:install/uninstall (note: some feature cannot be uninstalled in the current system)
  - catalog:ingest, catalog:removeall
  - log:tail, log:set <level> <logger>, log:get, log:list
  - using the up and down arrows on the console command line
- Confirm that installing via each install profile works.
- Confirm that logging still works and is still configurable from org.ops4j.pax.logging.cfg and from the Karaf console. Also, try switching the logging configuration to the xml-style configuration (see comment in org.ops4j.pax.logging.cfg, and confirm that logging still works and is configurable from log4j2.config.xml and from the Karaf console.
- Confirm SSHing into a running instance still works. Confirm SSHing into itests still works.
- Confirm debugging a live system and debugging itests still works.
- Test that install with ?dev=true and restart still works 
- Test running as a service and Automatic Start on System Boot (see Documentation)
#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4227](https://codice.atlassian.net/browse/DDF-4227)
#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
